src/update_collection_metadata.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+
+import github3
+from prefect import flow, task
+from prefect.blocks.system import Secret
+from prefect.deployments import run_deployment
+from prefect.states import Completed
+
+from utils import LatestReleases
+
+
+@task
+async def collection_needs_update(collection_name: str) -> bool:
+    """
+    Checks if the collection needs to be updated.
+    """
+    github_token = await Secret.load("github-token")
+    gh = github3.login(token=github_token.get())
+
+    collection_repo = gh.repository("PrefectHQ", collection_name)
+
+    latest_releases = await LatestReleases.load("collections")
+    latest_observed_release = latest_releases.get(collection_name)
+    latest_release = collection_repo.latest_release().tag_name
+
+    if latest_observed_release and latest_release == latest_observed_release:
+        print(
+            f"Collection {collection_name} is up to date! - "
+            f"(latest release: {latest_release})"
+        )
+        return False
+
+    latest_releases.set(collection_name, latest_release)
+    await latest_releases.save("collections", overwrite=True)
+
+    return True
+
+
+@task
+async def create_ref_if_not_exists(collection_name: str, branch_name: str):
+    """
+    Creates a reference to the latest release if it doesn't exist.
+    """
+    github_token = await Secret.load("github-token")
+    gh = github3.login(token=github_token.get())
+    repo = gh.repository("PrefectHQ", "prefect-collection-registry")
+
+    latest_release = (
+        gh.repository("PrefectHQ", collection_name).latest_release().tag_name
+    )
+
+    try:
+        repo.create_ref(
+            ref=f"refs/heads/{branch_name}",
+            sha=repo.commit(sha="main").sha,
+        )
```

Comment:
Wondering what this does

Reply:
creates a github branch to submit commits to

Reply:
Thanks for explaining. Can we add a comment here? The "main" sha is a tad concerning at first glance.

Reply:
i can add a comment, its just telling the new branch to start from main